packages/std/src/math/fraction.rs

```diff
@@ -12,3 +17,86 @@ pub trait Fraction<T>: Sized {
     /// If `p` is zero, None is returned.
     fn inv(&self) -> Option<Self>;
 }
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+pub struct Fraction<T>(T, T);
+
+impl<T> Fraction<T> {
```

Comment:
You only need this type for testing, right? Or do you need it also for messages and storage?

In order to avoid the naming collision, we could implement Fraction for the pair:

```
impl<T: Clone> Fraction<T> for (T, T) {
```